cmd/zpool/zpool_main.c

```diff
@@ -897,7 +897,8 @@ zpool_do_labelclear(int argc, char **argv)
 	if (zpool_read_label(fd, &config, NULL) != 0 || config == NULL) {
 		(void) fprintf(stderr,
 		    gettext("failed to check state for %s\n"), vdev);
-		return (1);
+		ret = 1;
+		goto errclose;
```

Comment:
This same leak also appears a few lines down in the `zpool_in_use()` block.  I'd suggest updating both blocks to look like the `default` case in a the switch statement below.  You can safely `goto errout`, when `name == NULL` free won't do anything so it's safe.  See the `free(3)` man page for details.

```
        ret = 1;
        goto errout;
```

Reply:
Yeah I know `free()` is safe to call. The point of adding the extra label was for documentation purposes plus to make it easier for the next guy to add an if statement and jump in the middle. i.e a clear ordering of statements to (reversed) labels. I'll remove the label and jump to `errout` though if you think its clearer. Will attend to the other leak also. Thanks

Reply:
I'd personally opt for jumping to `errout` but I'm OK with either approach as long as both leaks are addressed.

Reply:
Let's go with `errout`